probe-rs/src/architecture/arm/core/armv8m.rs

```diff
@@ -464,6 +464,63 @@ impl<'probe> CoreInterface for Armv8m<'probe> {
 
         Ok(())
     }
+
+    #[tracing::instrument(skip(self))]
+    fn enable_vector_catch(&mut self, condition: VectorCatchCondition) -> Result<(), Error> {
+        let mut dhcsr = Dhcsr(self.memory.read_word_32(Dhcsr::get_mmio_address())?);
+        dhcsr.set_c_debugen(true);
+        self.memory
+            .write_word_32(Dhcsr::get_mmio_address(), dhcsr.into())?;
+
+        let mut demcr = Demcr(self.memory.read_word_32(Demcr::get_mmio_address())?);
+        let idpfr1 = IdPfr1(self.memory.read_word_32(IdPfr1::get_mmio_address())?);
+        match condition {
+            VectorCatchCondition::HardFault => demcr.set_vc_harderr(true),
+            VectorCatchCondition::CoreReset => demcr.set_vc_corereset(true),
+            VectorCatchCondition::SecureFault => {
+                if !idpfr1.security_present() {
+                    return Err(Error::Arm(ArmError::ExtensionRequired(&["Security"])));
+                }
+                demcr.set_vc_sferr(true);
+            }
+            VectorCatchCondition::All => {
+                demcr.set_vc_harderr(true);
+                demcr.set_vc_corereset(true);
+                if idpfr1.security_present() {
+                    demcr.set_vc_sferr(true);
+                }
+            }
+        };
+
+        self.memory
+            .write_word_32(Demcr::get_mmio_address(), demcr.into())?;
+        Ok(())
+    }
+
+    fn disable_vector_catch(&mut self, condition: VectorCatchCondition) -> Result<(), Error> {
+        let mut demcr = Demcr(self.memory.read_word_32(Demcr::get_mmio_address())?);
+        let idpfr1 = IdPfr1(self.memory.read_word_32(IdPfr1::get_mmio_address())?);
+        match condition {
+            VectorCatchCondition::HardFault => demcr.set_vc_harderr(false),
+            VectorCatchCondition::CoreReset => demcr.set_vc_corereset(false),
+            VectorCatchCondition::SecureFault => {
+                if idpfr1.security_present() {
+                    demcr.set_vc_sferr(false);
+                }
```

Comment:
Any thoughts on silently continuing here if the security extension is not implemented, vs returning an error as I did for armv6m and armv7m?

Reply:
So I would do an evaluation how often this happens :)

If this is the general case that it silently continues, then maybe I would keep it so because it would fiddle with existing behavior. If it's the odd one out case, then I would definitely return an error.

Reply:
It seems like the general approach is to be returning an error. I'll make this change as well. This shouldn't break existing functionality as `VectorCatchCondition::SecureFault` is a new case.